### PR TITLE
Implement: unhide child pages from sitemap

### PR DESCRIPTION
**Pull request by reggie7**
Implements #546. I am adding a new property `excludeTreeFromSitemap` that works along `excludeFromSitemap`. They are independent now and allow to hide either the page alone or its subtree in site map. It requires content adjustment (I don't see a feasible solution that wouldn't). Requires [180](https://github.com/headwirecom/themeclean-flex/pull/180) to run properly.

### CODE REVIEW

File: platform/base/core/src/main/java/com/peregrine/sitemap/PageRecognizerBase.java [hunk 1]
```diff
@@ -37,17 +37,28 @@
...
 
     public final boolean isPage(final Page candidate) {
+        return isAPageBasically(candidate) && !isExcludedByProperty(excludeFromSiteMapPropertyName, candidate);
+    }
+
+    public final boolean isBucket(final Page candidate) {
+        return isAPageBasically(candidate) && !isExcludedByProperty(excludeTreeFromSiteMapPropertyName, candidate);
+    }
+
+    private boolean isAPageBasically(final Page candidate) {
```
Comment (reusr1) [inline platform/base/core/src/main/java/com/peregrine/sitemap/PageRecognizerBase.java]: I struggle with this method name - not sure what **isAPageBasically** means by reading that name

**Review by reusr1 — APPROVED** [review]
approving with the caveat that I really dont like isAPageBasically as a method name

File: platform/base/core/src/main/java/com/peregrine/sitemap/PageRecognizerBase.java (hunk 1 above)
Comment (reggie7) [reply platform/base/core/src/main/java/com/peregrine/sitemap/PageRecognizerBase.java]: I struggled with finding a good name indeed. I've come up with an alternative `hasAllPageMarkers`, hope will be good enough.

**Review by flx-sta — APPROVED** [review]
looks good and the naming issue seems to be resolved. The new one sounds a bit more explanatory

**Pull request merged**